server/lib/src/valueset/image.rs

```diff
@@ -101,14 +107,80 @@ enum ChunkStatus {
     MoreChunks,
 }
 
+fn png_consume_chunks_until_iend(buf: &[u8]) -> Result<(ChunkStatus, &[u8]), ImageValidationError> {
+    // length[u8;4] + chunk_type[u8;4] + checksum[u8;4] + minimum size
+    if buf.len() <= 8 {
+        return Err(ImageValidationError::InvalidImage(
+            "PNG file is too short to be valid".to_string(),
+        ));
+    } else {
+        #[cfg(any(debug_assertions, test))]
+        trace!("input buflen: {}", buf.len());
+    }
+    let (length_bytes, buf) = buf.split_at(4);
+    let (chunk_type, buf) = buf.split_at(4);
+
+    // Infallible: We've definitely consumed 4 bytes
+    let length = u32::from_be_bytes(
+        length_bytes
+            .clone()
+            .try_into()
+            .map_err(|_| ImageValidationError::InvalidImage("PNG corrupt!".to_string()))?,
+    );
+    #[cfg(any(debug_assertions, test))]
+    trace!(
+        "length_bytes: {:?} length: {} chunk_type: {:?} buflen: {}",
+        length_bytes,
+        &length,
+        &chunk_type,
+        &buf.len()
+    );
+
+    let (_, buf) = buf.split_at(length as usize);
```

Comment:
length here is input controlled, not controlled by us. So as a result, we need to validate at this point that length <= buf.len else split_at will panic.

Reply:
yep, un-pushed change has these checks 😄